File: Fauzan Risqullah-SMKN 4 Tangerang/rest/rest/FrmViewOrder.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Configuration;
using System.Data;
using System.Data.SqlClient;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace rest
{
    public partial class FrmViewOrder : Form
    {
        static string myconnstr = ConfigurationManager.ConnectionStrings["connstr"].ConnectionString;
        SqlConnection conn = new SqlConnection(myconnstr);

        public FrmViewOrder()
        {
            InitializeComponent();
        }

        private void FrmViewOrder_Load(object sender, EventArgs e)
        {
            string sql = "SELECT * FROM headerorder WHERE status = 'Memesan'";
            SqlCommand cmd = new SqlCommand(sql, conn);
            SqlDataAdapter sda = new SqlDataAdapter(cmd);

            DataTable dt = new DataTable();
            sda.Fill(dt);

            for (int i = 0; i < dt.Rows.Count; i++)
            {
                cmbID.Items.Add(dt.Rows[i]["orderid"].ToString());
            }
        }

        private void cmbID_SelectedIndexChanged(object sender, EventArgs e)
        {
            string sql = "SELECT * FROM detailorders WHERE orderid = '" + int.Parse(cmbID.Text) + "'";

            SqlCommand cmd = new SqlCommand(sql, conn);
            SqlDataAdapter sda = new SqlDataAdapter(cmd);

            DataTable dt = new DataTable();
            sda.Fill(dt);

            dgv.DataSource = dt;

            DataGridViewComboBoxColumn cmb = new DataGridViewComboBoxColumn();
            cmb.HeaderText = "Action";
            cmb.Name = "Combo";
            cmb.Items.Add("Memasak");
            cmb.Items.Add("Diantar");
            cmb.Items.Add("Pending");
            cmb.Items.Add("Selesai");

            dgv.Columns.Add(cmb);
        }

        private void dgv_SelectionChanged(object sender, EventArgs e)
        {
            
        }

        private void dgv_EditingControlShowing(object sender, DataGridViewEditingControlShowingEventArgs e)
        {

        }

        private void dgv_CellClick(object sender, DataGridViewCellEventArgs e)
        {

        }

        private void btnUpdate_Click(object sender, EventArgs e)
        {
            string sql = "UPDATE detailorders SET status = '" + dgv.Rows[0].Cells["Combo"].Value.ToString() + "' WHERE detailid = '"+ int.Parse(dgv.Rows[0].Cells[0].Value.ToString()) + "'";

            SqlCommand cmd = new SqlCommand(sql, conn);
            conn.Open();
            cmd.ExecuteNonQuery();
            conn.Close();

            string sql2 = "SELECT * FROM detailorders WHERE orderid = '"+int.Parse(cmbID.Text)+"'";
            SqlCommand cmd2 = new SqlCommand(sql2, conn);
            SqlDataAdapter sda2 = new SqlDataAdapter(cmd2);

            DataTable dt2 = new DataTable();
            sda2.Fill(dt2);

            dgv.Columns.Remove("Combo");
            dgv.DataSource = dt2;
            DataGridViewComboBoxColumn cmb = new DataGridViewComboBoxColumn();
            cmb.HeaderText = "Action";
            cmb.Name = "Combo";
            cmb.Items.Add("Memasak");
            cmb.Items.Add("Diantar");
            cmb.Items.Add("Pending");
            cmb.Items.Add("Selesai");

            dgv.Columns.Add(cmb);
        }

        private void button4_Click(object sender, EventArgs e)
        {
            this.Hide();
        }
    }
}

File: Daniel SMKS Bhaktianindya/SMKNUNSATAR/SMKNUNSATAR/FrmViewOrder.vb
Imports System.Data.SqlClient
Public Class FrmViewOrder
    Private Sub FrmViewOrder_Closed(sender As Object, e As EventArgs) Handles Me.Closed
        If CloseReason.ApplicationExitCall Then
            FrmChefNav.Show()
        End If
    End Sub

    Sub getorder()
        Call Koneksi()
        cmd = New SqlCommand("SELECT OrderID from Headerorder", conn)
        dr = cmd.ExecuteReader
        While dr.Read()
            Dim idorder As Integer = 0
            idorder = dr.Item(0)
            ComboBox1.Items.Add(idorder)
        End While
    End Sub

    Private Sub FrmViewOrder_Load(sender As Object, e As EventArgs) Handles Me.Load
        Call getorder()
        Call tampildata()
    End Sub

    Sub tampildata()
        Dim tampil As New DataGridViewColumn
        Call Koneksi()
        da = New SqlDataAdapter("SELECT Msmenu.Name,Detailorder.Qty From Detailorder inner join Msmenu on Msmenu.Menuid = Detailorder.Menuid where Orderid = '" & ComboBox1.Text & "' ", conn)
        ds = New DataSet
        da.Fill(ds, "Detailorder")
        DataGridView1.DataSource() = ds.Tables("Detailorder")
    End Sub

    Sub dd()
        Dim cc As New DataGridViewComboBoxColumn
        cc.HeaderText = "Action"
        cc.Name = "cc"
        cc.Items.AddRange("Pending,Deliver,Cooking")
        DataGridView1.Rows.Add(cc)
    End Sub
End Class